ddtrace/propagation/http.py

```diff
@@ -15,16 +23,279 @@
+        try:
+            return headers[header]
+        except KeyError:
+            pass
+
+    return default
+
+
+def _normalize_headers(headers):
+    # type: (Dict[str, str]) -> Dict[str, str]
+    return {name.lower(): v for name, v in headers.items()}
+
+
+def _b3_id_to_dd_id(b3_id):
+    # type: (str) -> int
+    """Helper to convert B3 trace/span hex ids into Datadog compatible ints
+
+    If the id is > 64 bit then truncate the trailing 64 bit.
+
+    "463ac35c9f6413ad48485a3953bb6124" -> "48485a3953bb6124" -> 5208512171318403364
+    """
+    if len(b3_id) > 16:
+        return int(b3_id[-16:], 16)
+    return int(b3_id, 16)
+
+
+def _dd_id_to_b3_id(dd_id):
+    # type: (int) -> str
+    """Helper to convert Datadog trace/span int ids into B3 compatible hex ids"""
+    # DEV: `hex(dd_id)` will give us `0xDEADBEEF`
+    # DEV: this gives us lowercase hex, which is what we want
+    return "{:x}".format(dd_id)
+
+
+def _inject_datadog(span_context, headers):
+    # type: (Context, Dict[str, str]) -> None
+    if span_context.trace_id is None or span_context.span_id is None:
+        return
+
+    headers[HTTP_HEADER_TRACE_ID] = str(span_context.trace_id)
+    headers[HTTP_HEADER_PARENT_ID] = str(span_context.span_id)
+    sampling_priority = span_context.sampling_priority
+    # Propagate priority only if defined
+    if sampling_priority is not None:
+        headers[HTTP_HEADER_SAMPLING_PRIORITY] = str(span_context.sampling_priority)
+    # Propagate origin only if defined
+    if span_context.dd_origin is not None:
+        headers[HTTP_HEADER_ORIGIN] = str(span_context.dd_origin)
+
+
+def _extract_datadog(headers):
+    # type: (Dict[str, str]) -> Optional[Context]
+    # TODO: Fix variable type changing (mypy)
+    trace_id = _extract_header_value(
+        POSSIBLE_HTTP_HEADER_TRACE_IDS,
+        headers,
+    )
+    if trace_id is None:
+        return None
+
+    parent_span_id = _extract_header_value(
+        POSSIBLE_HTTP_HEADER_PARENT_IDS,
+        headers,
+        default="0",
+    )
+    sampling_priority = _extract_header_value(
+        POSSIBLE_HTTP_HEADER_SAMPLING_PRIORITIES,
+        headers,
+    )
+    origin = _extract_header_value(
+        POSSIBLE_HTTP_HEADER_ORIGIN,
+        headers,
+    )
+
+    # Try to parse values into their expected types
+    try:
+        if sampling_priority is not None:
+            sampling_priority = int(sampling_priority)  # type: ignore[assignment]
+        else:
+            sampling_priority = sampling_priority
+
+        return Context(
+            # DEV: Do not allow `0` for trace id or span id, use None instead
+            trace_id=int(trace_id) or None,
+            span_id=int(parent_span_id) or None,  # type: ignore[arg-type]
+            sampling_priority=sampling_priority,  # type: ignore[arg-type]
+            dd_origin=origin,
+        )
+    # If headers are invalid and cannot be parsed, return a new context and log the issue.
```

Comment:
```suggestion
```

We don't return a new context and I think this comment is kinda unnecessary (code is obvious)